File: src/main/java/com/fifo/ticketing/global/util/UserValidator.java
package com.fifo.ticketing.global.util;

import com.fifo.ticketing.domain.user.dto.SessionUser;
import com.fifo.ticketing.global.exception.ErrorCode;
import com.fifo.ticketing.global.exception.ErrorException;
import jakarta.servlet.http.HttpSession;

public class UserValidator {

    private UserValidator() {
    }

    public static void validateSessionUser(HttpSession session) {
        SessionUser user = (SessionUser) session.getAttribute("loginUser");
        if (user == null) {
            throw new ErrorException("user/sign_in", ErrorCode.UNAUTHORIZED_REQUEST);
        }
    }
}

File: src/main/java/com/fifo/ticketing/global/util/UserValidator.kt
package com.fifo.ticketing.global.util

import com.fifo.ticketing.domain.user.dto.SessionUser
import com.fifo.ticketing.global.exception.ErrorCode
import com.fifo.ticketing.global.exception.ErrorException
import jakarta.servlet.http.HttpSession

object UserValidator {

    @JvmStatic
    fun validateSessionUser(session: HttpSession): SessionUser {
        val user = session.getAttribute("loginUser") as? SessionUser
            ?: throw ErrorException(ErrorCode.UNAUTHORIZED_REQUEST, "user/sign_in")
        return user
    }
}
